packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls.tsx

```diff
@@ -24,12 +25,34 @@ export const Title = (): React.JSX.Element => {
   return <TitleElement>{title}</TitleElement>;
 };
 
+const LocationDetailViewRefresh = () => {
+  const [{ history }] = useControl({
+    type: 'NAVIGATE',
+  });
+
+  const [{ data, isLoading }, handleList] = useAction({
+    type: 'LIST_LOCATION_ITEMS',
+  });
+
+  const prefix = history.join('');
+
+  return (
+    <Refresh
+      disabled={isLoading || data.result.length <= 0}
+      onClick={() =>
+        handleList({ prefix, options: { refresh: true, pageSize: 1000 } })
+      }
+    />
+  );
+};
+
 // @ts-expect-error TODO: add Controls assignment
 export const LocationDetailViewControls: LocationDetailViewControls = () => {
   return (
     <>
       <Navigate />
       <Title />
+      <LocationDetailViewRefresh />
```

Comment:
We could do something like this for now so they aren't overlapping in the UI. The spacing is weird but I think my grid PR will take care of that.
```
.storage-browser__refresh {
  grid-row-start: 2;
  grid-column: 26 / -1;
  justify-self: flex-end;
}

/* Targets when .storage-browser when it has .storage-browser__action-menu sibling */
.storage-browser__refresh:has(~ .storage-browser__action-menu) {
  grid-column: 26 / 30;
}
```